src/Microsoft.VisualStudio.Threading/JoinableTaskInternals.cs

```diff
@@ -20,4 +21,32 @@ public static bool IsMainThreadBlockedByAnyJoinableTask(JoinableTaskContext? joi
     {
         return joinableTaskContext?.IsMainThreadBlockedByAnyJoinableTask == true;
     }
+
+    public static JoinableTaskToken? GetJoinableTaskToken(JoinableTaskContext? joinableTaskContext)
+    {
+        if (joinableTaskContext?.AmbientTask?.WeakSelf is null)
```

Comment:
minor, maybe 
```csharp
if (joinableTaskContext?.AmbientTask?.WeakSelf is WeakReference<JoinableTask> reference)
{
   return new JoinableTaskToken() { JoinableTaskReference = reference };
}
```

so we never create the structure with null reference inside? or we can just do:

return new JoinableTaskToken() { JoinableTaskReference = joinableTaskContext?.AmbientTask?.WeakSelf }

And the structure has a function to check whether it is empty.  It feels that the nullable of the structure, and the inner value seems to be redundant.

Reply:
I added this check to avoid creating an entry in VS side when JoinableTaskToken is empty. Since the token's JoinableTaskReference is not available outside of vs-threading, I used the nullability of JoinableTaskToken to decide whether to create an entry for it.

Reply:
I understand. That is actually my suggestion:
1, we never return JoinableTaskToken with JoinableTaskReference is null. To do that, I think we capture the reference in a local variable, instead of accessing AmbientTask second time after checking the value. Because the task can be GCed between the two lines.

2, allow JoinableTaskReference to be null, and add IsNull or IsEmpty check to replace null check on the other side.

Maybe I prefer the first choice, so you don't have to update the code on the other side.

Reply:
Got it. Pushed an updated.